File: src/main/java/ru/citeck/ecos/history/dto/HistoryRecordDto.java
package ru.citeck.ecos.history.dto;

import lombok.Data;

import java.io.Serializable;

@Data
public class HistoryRecordDto implements Serializable {

    private static final long serialVersionUID = -8215328451338606217L;

    private String uuid;
    private String historyEventId;
    private String comments;
    private String version;
    private String username;
    private String userId;
    private String eventType;
    private Long creationTime;
    private String taskTitle;
    private String taskRole;
    private String taskOutcome;
    private String taskOutcomeName;
    private String taskDefinitionKey;
    private String taskType;
    private String documentId;

    public HistoryRecordDto() {
    }

    public HistoryRecordDto(HistoryRecordDto other) {
        this.uuid = other.uuid;
        this.historyEventId = other.historyEventId;
        this.comments = other.comments;
        this.version = other.version;
        this.username = other.username;
        this.userId = other.userId;
        this.eventType = other.eventType;
        this.creationTime = other.creationTime;
        this.taskTitle = other.taskTitle;
        this.taskRole = other.taskRole;
        this.taskOutcome = other.taskOutcome;
        this.taskOutcomeName = other.taskOutcomeName;
        this.taskDefinitionKey = other.taskDefinitionKey;
        this.taskType = other.taskType;
        this.documentId = other.documentId;
    }
}

File: src/main/java/ru/citeck/ecos/history/dto/HistoryRecordDto.kt
package ru.citeck.ecos.history.dto

import lombok.Data
import java.io.Serializable

@Data
class HistoryRecordDto : Serializable {

    companion object {
        private const val serialVersionUID = -8215328451338606217L
    }

    var uuid: String? = null
    var historyEventId: String? = null
    var comments: String? = null
    var version: String? = null
    var username: String? = null
    var userId: String? = null
    var eventType: String? = null
    var creationTime: Long? = null
    var taskTitle: String? = null
    var taskRole: String? = null
    var taskOutcome: String? = null
    var taskOutcomeName: String? = null
    var taskDefinitionKey: String? = null
    var taskType: String? = null
    var documentId: String? = null

    constructor() {}

    constructor(other: HistoryRecordDto) {
        uuid = other.uuid
        historyEventId = other.historyEventId
        comments = other.comments
        version = other.version
        username = other.username
        userId = other.userId
        eventType = other.eventType
        creationTime = other.creationTime
        taskTitle = other.taskTitle
        taskRole = other.taskRole
        taskOutcome = other.taskOutcome
        taskOutcomeName = other.taskOutcomeName
        taskDefinitionKey = other.taskDefinitionKey
        taskType = other.taskType
        documentId = other.documentId
    }
}
